02-04-domain-validations/internal/platform/storage/mysql/course_repository.go

```diff
@@ -37,3 +37,35 @@ func (r *CourseRepository) Save(ctx context.Context, course mooc.Course) error {
 
 	return nil
 }
+
+func (r *CourseRepository) SearchAll(ctx context.Context) ([]mooc.Course, error) {
```

Comment:
You can return `nil` instead of empty slices `[]mooc.Course{}` for those early returns when there's an error.

Reply:
Understood but, for example in your NewCourse function, why do you return an empty course and not a nil when there is an error?

> func NewCourse(id, name, duration string) (Course, error) {
> 	   idVO, err := NewCourseID(id)
> 	   if err != nil {
> 		   return Course{}, err
> 	   }